### ISSUE TITLE
Discuss akka.io URL structure

### ISSUE THREAD

**Issue by ennru** [akka/akka.github.com#569]
Right now we use

https://doc.akka.io/docs/akka/current/
https://doc.akka.io/api/akka/current/
https://doc.akka.io/japi/akka/current/

https://doc.akka.io/docs/alpakka/current/
https://doc.akka.io/api/alpakka/current/

This brings duplications in the URLs which might make search engines degrade site quality.

We could go with sub-domains or just a different URL hierarchy.

Migration to any new structure must be kept in mind.

**Comment by ennru** [comment]
I think we should consider changing it to

doc.akka.io/akka (redirect to current)
doc.akka.io/akka/current
doc.akka.io/akka/current/api
doc.akka.io/akka/current/japi

doc.akka.io/alpakka (redirect to current)
doc.akka.io/alpakka/current
doc.akka.io/alpakka/current/api

**Comment by patriknw** [comment]
looks neat, are there any drawbacks of changing this? such all existing links out there which will now be redirects?

**Comment by growthlabsmktg** [comment]
I too think that URL structure looks nice.

Some questions:

1. How much control do we have with individual pages having certain URLs, Page Titles, H1s?

For example, https://doc.akka.io/docs/akka/current/guide/introduction.html

It would be better if this had the following:
- https://doc.akka.io/docs/akka/current/guide/introduction-to-akka.html
- Page title looks good, it's Introduction to Akka - Akka Docs
- H1 also looks good as its currently, Introduction to Akka.

Obviously, this can't always be true, as in cases of very long titles that would make the URL extremely long. But as a general rule of thumb, this would be better.

2. Does the documentation tool being used have the ability to create and maintain a sitemap? I found this - https://doc.akka.io/sitemap.xml but I cannot tell if it was automatically generated. It certainly looks like its not an exhaustive list of pages in the documentation. It would be really good to have it map all of the pages in the current directory. We'd have to split it up though if it's [too big. ](https://support.google.com/webmasters/answer/75712?hl=en)

3. Can we remove the file extensions in the URLs?

This is a great first step, I'm interested to dive into the details a bit.

**Comment by theotown** [comment]
> I think we should consider changing it to
> 
> doc.akka.io/akka (redirect to current)
> doc.akka.io/akka/current
> doc.akka.io/akka/current/api
> doc.akka.io/akka/current/japi
> 
> doc.akka.io/alpakka (redirect to current)
> doc.akka.io/alpakka/current
> doc.akka.io/alpakka/current/api

Also doc.akka.io/akka-http/...

**Comment by raboof** [comment]
> Obviously, this can't always be true, as in cases of very long titles that would make the URL extremely long. But as a general rule of thumb, this would be better

This is possible: we would simply manually change the filename, all references to it, and add a redirect from the old to the new name. It is some work, but certainly can be done.

> Does the documentation tool being used have the ability to create and maintain a sitemap

Since the various subprojects can independently generate and upload docs, the sitemap cannot be generated as part of that process. We could have a separate script generating the sitemap, though.

> Can we remove the file extensions in the URLs?

So changing each `foo.html` to a `foo/index.html`? Nothing is impossible, but I suspect this would be quite a bit of work.

> Is it possible to host the documentation directly on the Akka site? rather than as a subdomain?

Those are populated using different technologies, so moving them together would mean migrating - could be done of course.

Perhaps we should turn each of those suggestions into separate issues and prioritize them based on cost/benefit?

**Comment by patriknw** [comment]
>> Can we remove the file extensions in the URLs?

> So changing each foo.html to a foo/index.html? Nothing is impossible, but I suspect this would be quite a bit of work.

I don’t think that the solution would be to create a directory for each page. The files would be without extension but a web server would add the html content type.

I think this would cause trouble for local browsing. I don’t want to do this unless it’s motivated to have very high benefit.

**Comment by growthlabsmktg** [comment]
> > > Can we remove the file extensions in the URLs?
> 
> > So changing each foo.html to a foo/index.html? Nothing is impossible, but I suspect this would be quite a bit of work.
> 
> I don’t think that the solution would be to create a directory for each page. The files would be without extension but a web server would add the html content type.
> 
> I think this would cause trouble for local browsing. I don’t want to do this unless it’s motivated to have very high benefit.



Sorry I didn't mean that. I am asking if we can remove the .html from the ends of the URLs.  I don't want to put each page into their own directories or anything like that.

**Comment by ennru** [comment]
I split out these new issues for discussion in separate tickets. Let's continue just the URL structure discussion here.

**Comment by raboof** [comment]
A little leaner still could be:

akka.io (frontpage)
akka.io/team (regular jekyll content)

akka.io/docs (current reference docs)
akka.io/docs/api (current scaladocs)
akka.io/docs/japi (current javadocs)
akka.io/docs/[version] (versioned reference docs)
akka.io/docs/[version]/api (versioned scaladocs)
akka.io/docs/[version]/japi (versioned javadocs)

akka.io/http (current akka-http reference docs)
akka.io/http/api (current akka-http scaladoc)
akka.io/http/japi (current akka-http javadoc)
akka.io/http/[version] (versioned akka-http reference docs)
akka.io/http/[version]/api (versioned akka-http scaladocs)
akka.io/http/[version]/japi (versioned akka-http javadocs)

(etc for other akka-* subprojects)

**Comment by ignasi35** [comment]
> akka.io/docs (current reference docs)
> akka.io/docs/api (current scaladocs)
> akka.io/docs/japi (current javadocs)

When browsing some versioned content I like seeing the version in the URL. I think the URL's you introduce could redirect to the appropriate version URL.

Similarly, I'd continue supporting `current` as a valid value or `[version]` but I'd turn it into a redirection to the actual version. I find it a bit confusing the following situation:

![Screenshot_2020-03-09_at_10_05_51](https://user-images.githubusercontent.com/762126/76198196-ed9f3980-61ed-11ea-8193-977310659f9e.png)

Summing up:

akka.io (frontpage)
akka.io/team (regular jekyll content)

akka.io/docs (redirect to the latest `version` below)
akka.io/docs/api (redirect to the latest `version` below)
akka.io/docs/japi (redirect to the latest `version` below)
akka.io/docs/[version] (versioned reference docs)
akka.io/docs/[version]/api (versioned scaladocs)
akka.io/docs/[version]/japi (versioned javadocs)

NOTE: In my proposal, `[version]` supports `current` which redirects to the latest numeric `version` value. So `akka.io/docs/japi` and `akka.io/docs/current/japi` behave identically.

--------------------------------------------------

Then, I'm not sure if this was a slip or intentional:

> akka.io/http (current akka-http reference docs)
> akka.io/http/api (current akka-http scaladoc)
> akka.io/http/japi (current akka-http javadoc)
> akka.io/http/[version] (versioned akka-http reference docs)
> akka.io/http/[version]/api (versioned akka-http scaladocs)
> akka.io/http/[version]/japi (versioned akka-http javadocs)

I think we should converge to `akka.io/<project-name>/` (`alpakka`, `akka-http`,...) instead of custom aliases. I suspect you are making this suggestion so fewer characters will impact SEO? Any other advantage in using `/http/` over `/akka-http/`?

**Comment by theotown** [comment]
On this: "I think we should converge to akka.io/<project-name>/ (alpakka,
akka-http,...) instead of custom aliases. I suspect you are making this
suggestion so fewer characters will impact SEO? Any other advantage in
using /http/ over /akka-http/?"

I think we should have /akka-http/ and /akka-streams/ specifically called
out in the URL. It's better for SEO. Ideally, we could also have
/akka-cluster/ and so on in there, but I understand that's not in the scope
of this conversation.

Oliver White
Chief Storyteller - Lightbend, Inc.
oliver@lightbend.com
@TheOTown <https://twitter.com/theotown>

CERTIFICATION
Get Certified in Reactive Architecture by Lightbend
<https://www.lightbend.com/certified-reactive-architect?utm_source=email&utm_medium=email-signature&utm_campaign=PL-2018-Lightbend-Certified-Reactive-Architect>
Validate your skills. Stand out from the crowd. Go Reactive.
<https://www.lightbend.com/>
